nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocument.java

```diff
@@ -45,21 +51,28 @@
     @Param(name = "name", required = false)
     protected String name;
 
+    /**
+     * @since 11.1
+     */
+    @Param(name = CoreEventConstants.RESET_LIFECYCLE, required = false)
```

Comment:
Worth adding a description of the new parameter, through the `description` field.

You could also update the `@Operation` description to state there is a new parameter to reset the lifecycle.

Reply:
With the new comments coming, actually this part is outdated:
*If the input document is a Folderish, all its content is copied recursively.*
Children will be copied anyway. The question now is:
**"do we copy the comments with the document ?"**
wdyt @kevinleturc

Reply:
Comments or not, the following sentence is still accurate:
> If the input document is a Folderish, all its content is copied recursively.

However, you're right, we need to handle the copy document/comments, will be another NXP ticket.

Reply:
Please don't use `CoreEventConstants.RESET_LIFECYCLE` for the value, even though it's the same string. Operation parameters are in a namespace of their own.